lib/ui/gallery_app_bar_widget.dart

```diff
@@ -107,6 +115,37 @@ class _GalleryAppBarWidgetState extends State<GalleryAppBarWidget> {
     );
   }
 
+  Future<dynamic> _renameAlbum(BuildContext context) async {
+    if (widget.type != GalleryAppBarType.owned_collection) {
+      return;
+    }
+    var result = await showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return ChangeCollectionNameDialog(name: appBarTitle);
+      },
+      barrierColor: Colors.black.withOpacity(0.85),
+    );
+    if (result == null) {
+      return;
+    }
+
+    final dialog = createProgressDialog(context, "changing name...");
```

Comment:
Could perhaps move this logic into `ChangeCollectionNameDialog` to reduce the logic on the app bar? Not sure.

Reply:
In future, during UX revamp, I was thinking of making this `ChangeCollectionNameDialog` a generic user input dialog, which can use reused. That's why I kept it separate for now.